core/src/actors/blocks_manager/handlers.rs

```diff
@@ -63,3 +67,24 @@ impl Handler<GetHighestBlockCheckpoint> for BlocksManager {
         }
     }
 }
+
+impl Handler<AddNewBlock> for BlocksManager {
+    type Result = ();
+
+    fn handle(&mut self, msg: AddNewBlock, _ctx: &mut Context<Self>) {
+        match self.process_new_block(msg.block) {
+            Ok(hash) => {
+                // Get sessions manager address
+                let sessions_manager_addr = System::current().registry().get::<SessionsManager>();
+
+                let items = vec![witnet_data_structures::chain::InvVector::Block(hash)];
+                sessions_manager_addr.do_send(Broadcast {
+                    command: AnnounceItems { items },
+                });
+            }
+            Err(_) => {
+                debug!("Block already exists");
```

Comment:
Hmmm, perhaps for the time being an `Err` returned by `process_new_block` means exactly that. But more errors could appear (`InvalidBlock`, `AlreadyPresentBlock`...). Perhaps the `process_new_block` method should return a `WitnetResult` with the `Err` type being defined like we do with the [`PeersError`](https://github.com/witnet/witnet-rust/blob/master/p2p/src/peers/error.rs)

Reply:
I created a simple `BlocksManagerError`, but it will be replaced in the future by a `InventoryError` when the `InventoryManager` is created.